data/transactions/logic/assembler.go

```diff
@@ -1062,62 +1067,68 @@ func (ops *OpStream) assemble(fin io.Reader) error {
 		if asmFunc != nil {
 			ops.trace("%3d: %s\t", ops.sourceLine, opstring)
 			ops.RecordSourceLine()
-			err := asmFunc(ops, &spec, fields[1:])
-			if err != nil {
-				return err
-			}
+			asmFunc(ops, &spec, fields[1:])
```

Comment:
Good point, will do.  I was thinking TEAL wouldn't be susceptible to the same concerns of a single error causing a cascading catastrophe because it's line-oriented.  But it's still probably the right thing to do.

Reply:
Since the assembly isn't going to be slow enough to matter, I decide the code will be more readable if I just proceed as now, but only report up to 10 errors.  So if I get cascading errors, the user experience will still be ok, but I don't need to insert checks at each error location to decide if I should quit now.

You comment made me revisit a number of places where I was returning early during assembling a line.  I found that I could give much better errors for future lines by "correcting" the error and continuing.  That way the internal type stack is maintained, even though you have a typo in your code, for example.  So thanks for that.